invenio_rdm_records/services/communities/service.py

```diff
@@ -37,10 +54,109 @@ def record_cls(self):
         """Factory for creating a record class."""
         return self.config.record_cls
 
+    def _exists(self, identity, community_id, record):
+        """Return the request id if an open request already exists, else None."""
+        results = current_requests_service.search(
+            identity,
+            extra_filter=dsl.query.Bool(
+                "must",
+                must=[
+                    dsl.Q("term", **{"receiver.community": community_id}),
+                    dsl.Q("term", **{"topic.record": record.pid.pid_value}),
+                    dsl.Q("term", **{"type": CommunityInclusion.type_id}),
+                    dsl.Q("term", **{"is_open": True}),
+                ],
+            ),
+        )
+        return next(results.hits)["id"] if results.total > 0 else None
+
+    def _include(self, identity, community_id, record, uow):
+        """Create request to add the community to the record."""
+        already_included = community_id in record.parent.communities
+        if already_included:
+            raise CommunityAlreadyExists()
+
+        # check if the community exists
+        community = current_communities.service.record_cls.pid.resolve(community_id)
+
+        # check if there is already an open request, to avoid duplications
+        existing_request_id = self._exists(identity, str(community.id), record)
+        if existing_request_id:
+            raise OpenRequestAlreadyExists(existing_request_id)
+
+        type_ = current_request_type_registry.lookup(CommunityInclusion.type_id)
+        receiver = ResolverRegistry.resolve_entity_proxy(
+            {"community": community_id}
+        ).resolve()
+
+        empty_data = {}
```

Comment:
not in the context of your PR, but it is a bit strange that we need to pass empty data, do we remember the reasons for it?

Reply:
Yes, to avoid extra comments in the conversation.